### ISSUE TITLE
Go-to definition in hover

### ISSUE THREAD

**Issue by MichaReiser** [astral-sh/ty#1575]
In rust-analyzer, I can click on `Command` to jump to its definition. 

<img width="595" height="315" alt="Image" src="https://github.com/user-attachments/assets/28156db4-d882-4f3e-8458-1d08a20e4946" />

We should also support this in ty

```py
from typing import TypedDict

class Test(TypedDict):
    name: str
    age: int



tests = [Test(), Test()]


for test in tests:
    pass
```

Hovering `test` reveals `Test`, it should be possible to click on `Test` and navigate to its definition

**Comment by dhruvmanila** [comment]
> In rust-analyzer, I can click on `Command` to jump to its definition.

But the code snippet that you've shared via a screenshot looks Python :)

**Comment by MichaReiser** [comment]
> > In rust-analyzer, I can click on `Command` to jump to its definition.
> 
> But the code snippet that you've shared via a screenshot looks Python :)

Yes. I was using ty while working on ty_benchmark and was very annoyed wheb I couldn't click on Command

**Comment by dhruvmanila** [comment]
Oh, sorry for being unclear, I thought you meant to provide a Rust code snippet demonstrating what it looks like in rust-analyzer. Is this what you're referring to in rust-analyzer? Specifically, the "Goto" links at the bottom of the hover window.

<img width="1017" height="467" alt="Image" src="https://github.com/user-attachments/assets/012406c6-33e9-40b3-9d9f-1d0f50b162aa" />

**Comment by Gankra** [comment]
Note the entire "Go to..." pane is 100% a non-standard extension of rust-analyzer. Which is to say, we would have to implement UI for it in the vscode extension and it would only work in the vscode extension. I agree it rules, I was sad when I discovered this last week.

https://github.com/rust-lang/rust-analyzer/blob/2efc80078029894eec0699f62ec8d5c1a56af763/docs/book/src/contributing/lsp-extensions.md#hover-actions

**Comment by MichaReiser** [comment]
That's indeed very disappointing
